build_daemon/lib/data/build_status.dart

```diff
@@ -44,6 +45,10 @@ abstract class DefaultBuildResult
   DefaultBuildResult._();
 }
 
+/// The group of [BuildResult]s for a single build done by the daemon.
+///
+/// Since the daemon can build multiple targets in parallel, the results are
+/// grouped together.
```

Comment:
Do clients get results for only the targets they registered or all targets?

Reply:
A client will only get results if they are interested in a target contained in the Build Result set. This may be an oversight and maybe we should update it to only return results of targets they are interested in.

Reply:
I don't think it is a big deal (although they only care about results for their targets presumably). But we should call out the behavior in the docs.